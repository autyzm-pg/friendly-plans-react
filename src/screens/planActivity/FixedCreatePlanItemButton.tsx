import React from 'react';
import { Animated, StyleSheet } from 'react-native';
import { FloatingAction } from 'react-native-floating-action';

import { Icon } from 'components';
import { i18n } from 'locale';
import { palette, typography } from 'styles';

const actionNames = {
  task: 'create-task',
  interaction: 'create-interaction',
  break: 'create-break',
};

const actions = [
  {
    name: actionNames.task,
    icon: <Icon name="layers" type="material" color={palette.primary} size={22} />,
    text: i18n.t('updatePlan:addTask'),
    position: 1,
    color: palette.background,
    textBackground: palette.primaryVariant,
    textStyle: typography.caption,
    textElevation: 0,
    textColor: palette.textWhite,
  },
  {
    name: actionNames.interaction,
    icon: <Icon name="group" type="material" color={palette.secondary} size={22} />,
    text: i18n.t('updatePlan:addInteraction'),
    position: 2,
    color: palette.background,
    textBackground: palette.primaryVariant,
    textStyle: typography.caption,
    textElevation: 0,
    textColor: palette.textWhite,
  },
  {
    name: actionNames.break,
    icon: <Icon name="notifications" type="material" color={palette.break} size={22} />,
    text: i18n.t('updatePlan:addBreak'),
    position: 3,
    color: palette.background,
    textBackground: palette.primaryVariant,
    textStyle: typography.caption,
    textElevation: 0,
    textColor: palette.textWhite,
  },
];

interface Props {
  onPress: (name: string) => void;
}

interface State {
  actionName: string;
  isOpen: boolean;
}

export class FixedCreatePlanItemButton extends React.PureComponent<Props, State> {
  state: State = {
    actionName: '',
    isOpen: false,
  };

  onPressItem = (actionName: string = actionNames.task) => {
    this.props.onPress(actionName);
  };

  onOpen = () => {
    this.setState({ isOpen: true });
  };

  onClose = () => {
    this.setState({ isOpen: false });
  };

  renderFloatingIcon = () => {
    return this.state.isOpen ? (
      <Icon name="close" type="material" color={palette.textWhite} size={32} />
    ) : (
      <Icon name="add" type="material" color={palette.secondary} size={32} />
    );
  };

  render() {
    return (
      <>
        {this.state.isOpen && <Animated.View style={[styles.overlay]} />}
        <FloatingAction
          color={this.state.isOpen ? palette.secondary : palette.primaryVariant}
          actions={actions}
          showBackground={false}
          onPressItem={this.onPressItem}
          floatingIcon={this.renderFloatingIcon()}
          onOpen={this.onOpen}
          onClose={this.onClose}
        />
      </>
    );
  }
}

const styles = StyleSheet.create({
  overlay: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: palette.modalBackgroundOverlay,
  },
});
