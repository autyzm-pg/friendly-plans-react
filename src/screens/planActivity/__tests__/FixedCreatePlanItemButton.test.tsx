import { shallow } from 'enzyme';
import React from 'react';
import { FixedCreatePlanItemButton } from '../FixedCreatePlanItemButton';

describe('FixedCreatePlanItemButton', () => {
  it('renders correctly', () => {
    const createPlanButton = shallow(<FixedCreatePlanItemButton onPress={jest.fn()} />);

    expect(createPlanButton).toMatchSnapshot();
  });
});
