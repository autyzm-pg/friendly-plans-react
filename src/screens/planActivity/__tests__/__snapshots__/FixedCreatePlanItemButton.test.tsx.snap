// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedCreatePlanItemButton renders correctly 1`] = `
<Fragment>
  <FloatingAction
    actions={
      Array [
        Object {
          "color": "#FFFFFF",
          "icon": <Icon
            color="#6654EB"
            name="layers"
            size={22}
            type="material"
          />,
          "name": "create-task",
          "position": 1,
          "text": "Add task",
          "textBackground": "#4F3DD5",
          "textColor": "rgba(255, 255, 255, 1)",
          "textElevation": 0,
          "textStyle": Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 12,
            "letterSpacing": 0,
          },
        },
        Object {
          "color": "#FFFFFF",
          "icon": <Icon
            color="#FE9DCA"
            name="group"
            size={22}
            type="material"
          />,
          "name": "create-interaction",
          "position": 2,
          "text": "Add interaction",
          "textBackground": "#4F3DD5",
          "textColor": "rgba(255, 255, 255, 1)",
          "textElevation": 0,
          "textStyle": Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 12,
            "letterSpacing": 0,
          },
        },
        Object {
          "color": "#FFFFFF",
          "icon": <Icon
            color="#FF984D"
            name="notifications"
            size={22}
            type="material"
          />,
          "name": "create-break",
          "position": 3,
          "text": "Add break",
          "textBackground": "#4F3DD5",
          "textColor": "rgba(255, 255, 255, 1)",
          "textElevation": 0,
          "textStyle": Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 12,
            "letterSpacing": 0,
          },
        },
      ]
    }
    actionsPaddingTopBottom={8}
    animated={true}
    buttonSize={56}
    color="#4F3DD5"
    dismissKeyboardOnPress={false}
    distanceToEdge={30}
    floatingIcon={
      <Icon
        color="#FE9DCA"
        name="add"
        size={32}
        type="material"
      />
    }
    iconColor="#fff"
    iconHeight={15}
    iconWidth={15}
    listenKeyboard={false}
    mainVerticalDistance={0}
    onClose={[Function]}
    onOpen={[Function]}
    onPressItem={[Function]}
    openOnMount={false}
    overlayColor="rgba(68, 68, 68, 0.6)"
    overrideWithAction={false}
    position="right"
    shadow={Object {}}
    showBackground={false}
    visible={true}
  />
</Fragment>
`;
